skeleton/src/main/java/com/dropbox/kaiken/skeleton/initializers/AppServicesInitializer.kt

```diff
@@ -0,0 +1,49 @@
+package com.dropbox.kaiken.skeleton.initializers
+
+import com.dropbox.common.inject.AppScope
+import com.squareup.anvil.annotations.ContributesTo
+import dagger.Module
+import dagger.Provides
+import dagger.multibindings.ElementsIntoSet
+
+/**
+ * Interface to use when you have a component which you want to be called
+ * immediately at app startup.  Be mindful anything done here will block the app from launching.
+ *
+ * @ContributesMultibinding(SdkScope::class)
+ * @SingleIn(SdkScope::class)
+ * class FooInitializer @Inject constructor(
+ *     private val foo: Foo,
+ *     private val bar: Bar,
+ * ): AppInitializer {
+ *    override fun init() {
+ *       foo.setup(bar)
+ *    }
+ * }
+ *
+ */
+interface AppServicesInitializer {
+    /**
+     * Called when app services is initialized
+     */
+    fun init()
+
+    /**
+     * Called when the app is torn down
+     */
+    fun teardown()
```

Comment:
@rodgavila - I couldn't find a call to an app services level `teardown` method. Do you recall where in the app lifecycle this lived?

Reply:
Unfortunately, I don't think there's a good way for us to know when the app is being teardown. So this might actually not be possible.